sdk/core/azure-core/src/http/log_policy.cpp

```diff
@@ -30,22 +30,63 @@ std::string TruncateIfLengthy(std::string const& s)
   return s.substr(0, BeginLength) + Ellipsis + s.substr(length - EndLength, EndLength);
 }
 
-std::string GetRequestLogMessage(Request const& request)
+std::string GetRequestLogMessage(
+    Azure::Core::Http::LogOptions const& options,
+    Request const& request)
 {
   std::ostringstream log;
 
-  log << "HTTP Request : " << HttpMethodToString(request.GetMethod()) << " "
-      << request.GetUrl().GetAbsoluteUrl();
+  auto const& requestUrl = request.GetUrl();
+
+  std::string url = requestUrl.GetUrlWithoutQuery();
 
-  for (auto header : request.GetHeaders())
   {
-    log << "\n\t" << header.first << " : " << TruncateIfLengthy(header.second);
+    auto separ = '?';
+    for (auto const& qparam : requestUrl.GetQueryParameters())
+    {
+      url += separ + qparam.first + '=';
+
+      if (options.AllowedHttpQueryParameters.find(qparam.first)
+          != options.AllowedHttpQueryParameters.end())
+      {
+        url += TruncateIfLengthy(qparam.second);
+      }
+      else
+      {
+        url += "[hidden]";
```

Comment:
Are other languages adding this `hidden` placeholder?  or are they just not logging the header at all?  We could try to follow the same approach here.

Reply:
Thank you, I'll unify it.
.NET says "REDACTED" - https://github.com/Azure/azure-sdk-for-net/blob/41a7cf2cca7badc4be320ce57aa8b47adcdff137/sdk/core/Azure.Core/src/Shared/HttpMessageSanitizer.cs